object.c

```diff
@@ -3374,33 +3378,35 @@ rb_convert_to_integer(VALUE val, int base, int raise_exception)
 {
     VALUE tmp;
 
+    if (base) {
+        tmp = rb_check_string_type(val);
```

Comment:
Strictly speaking, this changes the behavior when `to_str` is defined to `Float`, `Integer`, and `NilClass`, but probably it does not matter, I guess.

Reply:
Yes I think nobody does that.